src/Migrations/M034MigrateCoversContentBlockToPostsListBlock.php

```diff
@@ -0,0 +1,465 @@
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+use WP_Block_Parser;
+
+/**
+ * Migrate Covers block of type Content to Posts List blocks.
+ */
+class M034MigrateCoversContentBlockToPostsListBlock extends MigrationScript
+{
+    /**
+     * Perform the actual migration.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    protected static function execute(MigrationRecord $record): void
+    {
+        try {
+            // Get the list of posts using Covers blocks.
+            $posts = Utils\Functions::get_posts_using_specific_block(
+                Utils\Constants::BLOCK_COVERS,
+                Utils\Constants::ALL_POST_TYPES,
+                Utils\Constants::POST_STATUS_LIST,
+            );
+
+            // If there are no posts, abort.
+            if (!$posts) {
+                return;
+            }
+
+            echo "Covers content block migration in progress...\n"; // phpcs:ignore
+
+            $parser = new WP_Block_Parser();
+
+            foreach ($posts as $post) {
+                if (empty($post->post_content)) {
+                    continue;
+                }
+
+                $current_post_id = $post->ID; // Store the current post ID
+
+                echo 'Parsing post ', $current_post_id, "\n"; // phpcs:ignore
+
+                // Get all the blocks of each post.
+                $blocks = $parser->parse($post->post_content);
+
+                if (!is_array($blocks)) {
+                    throw new \Exception("Invalid block structure for post #" . $current_post_id);
+                }
+
+                foreach ($blocks as &$block) {
+                    // Check if the block is valid.
+                    if (!is_array($block)) {
+                        continue;
+                    }
+
+                    // Check if the block has a 'blockName' key.
+                    if (!isset($block['blockName'])) {
+                        continue;
+                    }
+
+                    // Check if the block is a Cover block. If not, abort.
+                    if ($block['blockName'] !== Utils\Constants::BLOCK_COVERS) {
+                        continue;
+                    }
+
+                    // Check if the Cover block type is Content. If not, abort.
+                    // Cover blocks of type content have as value of $block['attrs']['cover_type']
+                    // the following possibilities: "content", "1", NULL
+                    // https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/26b480a0954667a0813ca8c3a90377f2a1fbea1c/classes/blocks/class-covers.php#L35
+                    // https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/26b480a0954667a0813ca8c3a90377f2a1fbea1c/classes/blocks/class-covers.php#L191
+                    $type = $block['attrs']['cover_type'];
```

Comment:
In case of default content cover type, the cover_type may not set and above line will throw -
`Warning: Undefined array key "cover_type"`

Reply:
Done!